src/reader.rs

```diff
@@ -414,7 +420,29 @@ impl<'a> Reader<'a> {
             return Err(Error::UnalignedBlob);
         }
 
-        let header = Reader::get_header(blob)?;
+        let header = Reader::get_header(blob, true)?;
+        Ok(Reader::<'a> {
+            reserved_mem: Reader::get_reserved_mem(blob, &header)?,
+            struct_block: Reader::get_struct_block(blob, &header)?,
+            strings_block: Reader::get_strings_block(blob, &header)?,
+        })
+    }
+
+    /// Reads a DTB blob given its memory address and returns a corresponding reader.
+    pub fn read_from_memory_address(addr: usize) -> Result<Self> {
```

Comment:
1. I would give it a shorter name, e.g. `read_from_addr()`.
2. This function has to be marked as `unsafe`.
3. I think we should move the `total_size` check out of `get_header()` not to litter with `known_size`.

Reply:
1. `read_from_address` maybe?
2. ok
3. I tried to touch as little as possible. Will look if moving this makes code nicer.
4. will add an UT

Reply:
1. ok